libs/designsystem/src/lib/components/popover/popover.component.ts

```diff
@@ -0,0 +1,210 @@
+    this.renderer.appendChild(this.document.body, this.elementRef.nativeElement);
+
+    const popoverElement = this.wrapperElement.nativeElement;
+    const popoverDimensions = popoverElement.getBoundingClientRect();
+    const viewport = popoverElement.ownerDocument.defaultView;
+    this.positionHorizontally(viewport, targetDimensions, popoverDimensions);
+    this.positionVertically(viewport, targetDimensions, popoverDimensions);
+  }
+
+  private positionHorizontally(
+    viewPort: Window,
+    targetDimensions: DOMRect,
+    popoverDimensions: DOMRect
+  ) {
+    const viewPortWidth = viewPort.innerWidth;
+    const popoverElement = this.wrapperElement.nativeElement;
+    const contentWidth = popoverDimensions.width;
+    const availableSpaceRight = viewPortWidth - targetDimensions.left;
+    const availableSpaceLeft = targetDimensions.right;
+    const contentCanRightOfTarget = availableSpaceRight >= contentWidth + this.POPOVER_BODY_PADDING;
+    const openRight = contentCanRightOfTarget || availableSpaceRight >= availableSpaceLeft;
+    if (this.popout === HorizontalDirection.left || !openRight) {
+      // Open left:
+      const rightPxValue = `${viewPortWidth - targetDimensions.right}px`;
+      this.renderer.removeStyle(popoverElement, 'left');
+      this.renderer.setStyle(popoverElement, 'right', rightPxValue);
+    } else {
+      // Open right:
+      const leftPxValue = `${targetDimensions.left}px`;
+      this.renderer.removeStyle(popoverElement, 'right');
+      this.renderer.setStyle(popoverElement, 'left', leftPxValue);
+    }
+  }
+
+  private positionVertically(
+    viewPort: Window,
+    targetDimensions: DOMRect,
+    popoverDimensions: DOMRect
+  ) {
+    const viewPortHeight = viewPort.innerHeight;
+    const popoverElement = this.wrapperElement.nativeElement;
+    const contentHeight = popoverDimensions.height;
+    const availableSpaceDown = viewPortHeight - targetDimensions.bottom;
+    const availableSpaceUp = targetDimensions.top;
+    const contentCanFitBelowTarget =
+      availableSpaceDown >= contentHeight + this.POPOVER_BODY_PADDING;
+    const openDown = contentCanFitBelowTarget || availableSpaceDown >= availableSpaceUp;
+    if (openDown) {
+      // Open down:
+      const topPxValue = `${targetDimensions.bottom}px`;
+      this.renderer.removeStyle(popoverElement, 'bottom');
+      this.renderer.setStyle(popoverElement, 'top', topPxValue);
+    } else {
+      // Open up:
+      const bottomPxValue = `${viewPortHeight - targetDimensions.top}px`;
+      this.renderer.removeStyle(popoverElement, 'top');
+      this.renderer.setStyle(popoverElement, 'bottom', bottomPxValue);
+      // Ensure target element is elevated above any shadow inside popover, i.e. content wrapped in Card:
+      this.renderer.setStyle(this.targetElement, 'z-index', `${this.zIndex + 1}`);
+      this.renderer.setStyle(this.targetElement, 'pointer-events', 'none');
+    }
+  }
+}
```

Comment:
Introducing popover with dropdown has created a bug in safari where the element is not selected when using the mouse. The arrow keys works perfectly well.

Reply:
Resolving this, take a look at this instead: https://github.com/kirbydesign/designsystem/pull/1671#discussion_r692033353